tests/commands/test_commands.py

```diff
@@ -392,24 +392,67 @@ def mock_write(tx, mode):
     args = Namespace(action='new',
                      public_key='CJxdItf4lz2PwEf4SmYNAu/c/VpmX39JEgC5YpH7fxg=',
                      power=1,
-                     node_id='12345',
+                     node_id='fb7140f03a4ffad899fabbbf655b97e0321add66',
                      sk=priv_validator_path,
                      config={})
-    resp = run_upsert_validator_new(args, b)
 
-    assert b.get_transaction(resp)
+    with caplog.at_level(logging.INFO):
+        election_id = run_upsert_validator_new(args, b)
+        assert caplog.records[0].msg == '[SUCCESS] Submitted proposal with id: ' + election_id
+        assert b.get_transaction(election_id)
+
+
+@pytest.mark.tendermint
+@pytest.mark.bdb
+def test_upsert_validator_new_invalid_election(caplog, b, priv_validator_path, user_sk):
+    from bigchaindb.commands.bigchaindb import run_upsert_validator_new
+
+    args = Namespace(action='new',
+                     public_key='CJxdItf4lz2PwEf4SmYNAu/c/VpmX39JEgC5YpH7fxg=',
+                     power=10,
+                     node_id='fb7140f03a4ffad899fabbbf655b97e0321add66',
+                     sk='/tmp/invalid/path/key.json',
+                     config={})
+
+    with caplog.at_level(logging.ERROR):
```

Comment:
If `run_upsert_validator_new` returns an error code (can be stored inside the exception class), asserting is as simple as:

```
assert run_upsert_validator_new(args, b) == SomeException.error_code
```

It is then easier to read and comprehend.

Though technically, the current approach is good too.

Reply:
Off course - the comprehensive output is important. It's just a little cumbersome to assert logs in tests. But I do not insist.